lib/ui/src/settings/index.tsx

```diff
@@ -1,84 +1,94 @@
-    height: 100%;
-  }
-`;
-
-interface PureSettingsPagesProps {
-  activeTab: string;
-  changeTab: (tab: string) => {};
-  onClose: () => {};
-}
-
-const PureSettingsPages: FunctionComponent<PureSettingsPagesProps> = ({
-  activeTab,
-  changeTab,
-  onClose,
-}) => (
-  <Wrapper>
-    <Tabs
-      absolute
-      selected={activeTab}
-      actions={{ onSelect: changeTab }}
-      tools={
-        <IconButton
-          onClick={(e: SyntheticEvent) => {
+const TabBarButton: FunctionComponent<{ id: string; title: string }> = ({ id, title }) => (
+  <Location>
+    {({ navigate, path }) => {
+      const active = path.includes(`settings/${id}`);
+      return (
+        <TabButton
+          id={`tabbutton-${id}`}
+          className={['tabbutton'].concat(active ? ['tabbutton-active'] : []).join(' ')}
+          type="button"
+          key="id"
+          active={active}
+          onClick={(e: any) => {
             e.preventDefault();
-            return onClose();
+            navigate(`/settings/${id}`);
           }}
+          role="tab"
         >
-          <Icons icon="close" />
-        </IconButton>
-      }
-    >
-      <div id={ABOUT} title="About">
-        <Route path={ABOUT}>
-          <AboutPage key={ABOUT} onClose={onClose} />
-        </Route>
-      </div>
+          {title}
+        </TabButton>
+      );
+    }}
+  </Location>
+);
+
+const Content = styled(ScrollArea)({
+  position: 'absolute',
+  top: 40,
+  left: 0,
+  right: 0,
+  bottom: 0,
+  overflow: 'auto',
+});
 
-      <div id={RELEASE_NOTES} title="Release notes">
-        <Route path={RELEASE_NOTES}>
-          <ReleaseNotesPage key={RELEASE_NOTES} onClose={onClose} />
-        </Route>
-      </div>
+const keyMap = {
+  CLOSE: 'escape',
+};
 
-      <div id={SHORTCUTS} title="Keyboard shortcuts">
-        <Route path={SHORTCUTS}>
-          <ShortcutsPage key={SHORTCUTS} onClose={onClose} />
-        </Route>
-      </div>
-    </Tabs>
-  </Wrapper>
+const Pages: FunctionComponent<{ onClose: API['closeSettings'] }> = ({ onClose }) => (
+  <Fragment>
+    <FlexBar border>
+      <TabBar role="tablist">
+        <TabBarButton id="about" title="About" />
+        <TabBarButton id="shortcuts" title="Keyboard shortcuts" />
+        <TabBarButton id="release-notes" title="Release notes" />
+      </TabBar>
+      <IconButton
+        onClick={(e: SyntheticEvent) => {
+          e.preventDefault();
+          return onClose();
+        }}
+      >
+        <Icons icon="close" />
+      </IconButton>
+    </FlexBar>
+    <Content vertical horizontal={false}>
+      <Route path="about">
+        <AboutPage key="about" />
+      </Route>
+      <Route path="release-notes">
+        <ReleaseNotesPage key="release-notes" />
+      </Route>
+      <Route path="shortcuts">
+        <ShortcutsPage key="shortcuts" />
+      </Route>
+    </Content>
+    <Route path="settings">
+      <GlobalHotKeys handlers={{ CLOSE: onClose }} keyMap={keyMap} />
+    </Route>
```

Comment:
Why wrap this in a `Route`?

Reply:
to optionally render that section based on location / route

Reply:
Right, I will remove this, good call!